### DISCUSSION TITLE
Weird attribute names

### DISCUSSION THREAD

**Discussion by home409ca** [BJReplay/ha-solcast-solar#138] [Q&A] [answered]
Solcast PV Forecast v 4.1.3
What are these attributes? In previous versions they were described, but here I see strange numbers...
![image](https://github.com/user-attachments/assets/d6237920-ca5c-4464-9b22-a1ea564f84d6)

**Comment by autoSteve** [comment]
These are a breakdown of Solcast sites. The 'strange numbers' are your rooftop site IDs.

If you don't want them, then turn them off in the integration configuration. There are options to do so to reduce clutter.

**Comment by autoSteve** [comment]
(If the translations to do so are inadequate do please call this out. Polish is not one of my languages.)

(And with the help of Google, Jeżeli tłumaczenia nie są wystarczające, proszę to zgłosić. Polski nie jest jednym z moich języków.)

**Comment by home409ca** [comment]
I created a translation, restarted HA but I still have the same...
![image](https://github.com/user-attachments/assets/b676c429-bf3d-4b47-9b88-b7100d4e9167)

[pl.json](https://github.com/user-attachments/files/16636788/pl.json)

**Comment by autoSteve** [comment]
I absolutely do not understand, @home409ca, but thanks for the translation.

The attributes are selectable when you configure the integration. Turn them off. Turn them on. The integration does not care. It will present attributes that are selected.

Atrybuty można wybrać podczas konfigurowania integracji. Wyłącz je. Włącz je. Ustawienie integracji nie ma znaczenia.  Pokaże wybrane atrybuty.

**Comment by autoSteve** [comment]
Go to ustawienia | urządzeń i usług | Solcast PV Forecast | KONFIGURUJ, then select atrybuty, then turn off those you do not need... Wszystko zostanie rozwiązane.

**Comment by autoSteve** [comment]
(I am quite new to Polish. Forgive...)

**Comment by home409ca** [comment]
I removed the HACS integration. I downloaded it again. I reinstalled and configured it again and I still see the Resource ID instead of the patch name.

![image](https://github.com/user-attachments/assets/dc979f80-f138-42b1-a9dd-ae9050230f31)

I am also sending the language file again, there was one error.

[pl.json](https://github.com/user-attachments/files/16640358/pl.json)

**Reply by autoSteve** [reply]
Thanks for the updated translation.

That diagnosic information looks fine to me. It's showing the names of your two rooftop sites, as configured at solcast.com. PV1/2 East/North?

Now that you have configured the available attributes (by turning off the sites breakdown) is that solved for you?

**Reply by home409ca** [reply]
No. I still have the same problem. I've redownloaded the add-on, turned off the attributes and turned them back on, but I still have the ID instead of the name...

**Reply by autoSteve** [reply]
I can't see any issue with that diagnostic card. It should show the site names, not IDs and that is what it is doing. An ID looks like this: "83d5-ac72-2a8a-2397", and a site name "PV1 Wschod". Looks perfect. This is mine:

<img width="298" alt="image" src="https://github.com/user-attachments/assets/2c144ec3-fdbd-4bfe-84cc-1110053490b2">

What do the attributes look like now with site breakdown turned off?

<img width="739" alt="image" src="https://github.com/user-attachments/assets/0673f60b-b66c-49d0-b1f5-32d9dbba55cb">

**Reply by autoSteve** [reply]
I just had a thought.

Are you expecting the names 'Estimate', 'Estimate10' and 'Estimate90' to be translated? If so, then that won't happen. They are attributes of sensors, and not sensors themselves.

**Reply by home409ca** [reply]
![image](https://github.com/user-attachments/assets/97300516-a53a-431e-a199-e8106502aaba)

I expect to see:
![image](https://github.com/user-attachments/assets/01901272-9dc7-4ffb-bdf0-35a01e5518f3)

**Reply by autoSteve** [reply, marked as answer]
Ahh.

I made a design decision to use the site ID and not the name.

The site ID will never change, but the site name could be changed, and that would result in things breaking and needing to be fixed in HA.

If you do actually want to use those attributes, then you could create template sensors using any name you wish that get required attributes using something like `{{ state_attr('sensor.solcast_pv_forecast_peak_forecast_today', 'estimate10-1234-5678-9012-3456') | float(0) }}`.
